db/schema_tables.cc

```diff
@@ -1285,10 +1285,10 @@ std::vector<mutation> make_drop_keyspace_mutations(schema_features features, lw_
  *
  * @param partition Keyspace attributes in serialized form
  */
-future<lw_shared_ptr<keyspace_metadata>> create_keyspace_from_schema_partition(distributed<service::storage_proxy>& proxy, const schema_result_value_type& result)
+future<lw_shared_ptr<keyspace_metadata>> create_keyspace_from_schema_partition(distributed<service::storage_proxy>& proxy,
+                                                                               const schema_result_value_type& result,
+                                                                               lw_shared_ptr<query::result_set> scylla_specific_rs)
 {
-    auto scylla_specific_rs = co_await extract_scylla_specific_keyspace_info(proxy, result);
```

Comment:
extract_scylla_specific_keyspace_info is now not used anywhere?

this doesn't seem correct, before during parse_system_tables it was called, now you pass nullptr there. I suspect this may be the reason why after restart new keyspace fallbacks to non-s3 storage, because scylla_specific_rs is not loaded

Reply:
Right, fixed, thanks. That indeed was the reason tests failed.
The function extract_scylla_specific_keyspace_info is used in `parse_system_tables` now.